### PR TITLE
Integrate API Error model and controller

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

File: crowbar_framework/app/controllers/api/errors_controller.rb [hunk 1]
```diff
@@ -15,29 +15,60 @@
...
   end
 
   api :POST, "/api/errors", "Create an error"
   api_version "2.0"
   def create
-    head :not_implemented
+    @error = Api::Error.new(error_params)
+
+    if @error.save
+      render json: @error, status: :ok
```
Comment (MaximilianMeister) [inline crowbar_framework/app/controllers/api/errors_controller.rb]: you can remove the `status: ok`, as it is the implicit default

File: crowbar_framework/app/controllers/api/errors_controller.rb [hunk 2]
```diff
@@ -15,29 +15,60 @@
...
   api :DELETE, "/api/errors/:id", "Delete a specific error"
   param :id, Integer, desc: "Error ID", required: true
   api_version "2.0"
   def destroy
-    head :not_implemented
+    if @error.destroy
+      head :ok
+    else
+      render json: {
+        error: "Failed to delete error"
```
Comment (MaximilianMeister) [inline crowbar_framework/app/controllers/api/errors_controller.rb]: I18n

File: crowbar_framework/app/models/api/error.rb [hunk 3]
```diff
@@ -14,7 +14,18 @@
...
+
+  validates :error,
+    presence: true
+
+  validates :message,
+    presence: true
+
+  validates :code,
+    inclusion: {
+      in: Rack::Utils::HTTP_STATUS_CODES.keys.concat([nil]),
```
Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/error.rb]: I wouldn't restrict to only http codes here, as we may want to save zypper codes, shell return codes etc.

**Comment by MaximilianMeister** [comment]
and can you add some model specs? at least for `save` and `destroy`?

File: crowbar_framework/config/locales/crowbar/en.yml [hunk 4]
```diff
@@ -845,6 +845,7 @@ en:
     chef_server_down: 'Chef server is not available'
     param_missing: 'A required parameter is missing'
     before_install: 'This API is not available before installation'
+    destroy_failed: 'Failed to destroy %{component}'
```
Comment (MaximilianMeister) [inline crowbar_framework/config/locales/crowbar/en.yml]: this should be in:
```yaml
api:
  errors:
    destroy_failed: 'Failed to destroy %{component}'
```

File: crowbar_framework/app/controllers/api/errors_controller.rb [hunk 5]
```diff
@@ -15,29 +15,59 @@
...
   api :DELETE, "/api/errors/:id", "Delete a specific error"
   param :id, Integer, desc: "Error ID", required: true
   api_version "2.0"
   def destroy
-    head :not_implemented
+    if @error.destroy
+      head :ok
+    else
+      render json: {
+        error: I18n.t("error.destroy_failed", component: "error")
```
Comment (MaximilianMeister) [inline crowbar_framework/app/controllers/api/errors_controller.rb]: and here `I18n.t("api.errors.destroy_failed", component: "error")`

**Comment by jsuchome** [comment]
+1

**Comment by rsalevsky** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;sKTTY1Wp&#x2F;65-3-s46p13-implement-api-for-first-step-of-upgrade-checks"> (3) S46P13: Implement API for first step of upgrade: checks</a>

**Comment by nicolasbock** [comment]
+1

**Comment by rsalevsky** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;sKTTY1Wp&#x2F;65-3-s46p13-implement-api-for-first-step-of-upgrade-checks"> (3) S46P13: Implement API for first step of upgrade: checks</a>

**Pull request merged**